### PR TITLE
feat: add Dockerfile extension

### PR DESCRIPTION
**Pull request by steschwa**
Icon for `*.Dockerfile` files

### CODE REVIEW

File: lua/nvim-web-devicons.lua [hunk 1]
```diff
@@ -1656,6 +1656,12 @@ local icons_by_file_extension = {
     cterm_color = "106",
     name = "Liquid",
   },
+  ["Dockerfile"] = {
```
Comment (alex-courtis) [inline lua/nvim-web-devicons.lua:1659]: We don't use specific cases for the filenames or extensions as `get_icon` is not case sensitive.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
What's the use case for this one? Is a `Dockerfile` not correctly applying somewhere?

**Comment by steschwa** [comment]
> What's the use case for this one? Is a `Dockerfile` not correctly applying somewhere?

If you want to build multiple docker images using the same context. Here's an example:

image

I am building an API that is distributed as a docker image. I therefore have a `Dockerfile`. To run database migrations i build another docker image. Because `Dockerfile` is already used i need a different filename `migrations.Dockerfile`. There is no official recommendation (as far as i am concerned) what naming convention to use to name multiple Dockerfiles. I found this SO https://stackoverflow.com/questions/26077543/how-to-name-dockerfiles that suggests naming them with the `.Dockerfile` extension.

File: lua/nvim-web-devicons.lua (hunk 1 above)
Comment (steschwa) [reply lua/nvim-web-devicons.lua:1659]: This is interesting. Currently i have the following configuration for `nvim-web-devicons` (notice the uppercase "Dockerfile"):

```lua
override_by_extension = {
    ["Dockerfile"] = {
        icon = "󰡨",
        color = "#458ee6",
        cterm_color = "68",
        name = "Dockerfile",
    },
},
```

if i chance it to be lowercased "dockerfile" the below example with `migrations.Dockerfile` doesn't get an icon

**Comment by alex-courtis** [comment]
> Because `Dockerfile` is already used i need a different filename `migrations.Dockerfile`. 

I understand. I did not note that it's an extension rather than a name.

We also need to handle the case of `Dockerfile.foo` which is a pattern used in the wild.

```
dockerfile
Dockerfile
dockerfile.bar
Dockerfile.bar
foo.dockerfile
foo.Dockerfile
foo.dockerfile.bar
foo.Dockerfile.bar
```

**Review by alex-courtis — APPROVED** [review]
Many thanks for your contribution!

**Pull request merged**